src/opencage.rs

```diff
@@ -110,49 +135,89 @@ impl Opencage {
     ///
-    /// You may restrict the search space by passing an optional bounding box to search within.
+    /// it is recommended that you restrict the search space by passing a
+    /// [bounding box](struct.InputBounds.html) to search within.
+    /// If you don't need or want to restrict the search using a bounding box (usually not recommended), you
+    /// may pass the [`NOBOX`](static.NOBOX.html) static value instead.
+    ///
     /// Please see [the documentation](https://opencagedata.com/api#ambiguous-results) for details
     /// of best practices in order to obtain good-quality results.
     ///
     /// This method passes the `no_record` parameter to the API.
     ///
+    /// # Examples
+    ///
     ///```
     /// use geocoding::{Opencage, Point};
     /// use geocoding::opencage::InputBounds;
     ///
     /// let oc = Opencage::new("dcdbf0d783374909b3debee728c7cc10".to_string());
     /// let address = "UCL CASA";
-    /// // restrict the search space. An `into()` conversion exists for `Point` tuples
-    /// let bbox = (
+    /// // Optionally restrict the search space using a bounding box.
+    /// // The first point is the bottom-left corner, the second is the top-right.
+    /// let bbox = InputBounds::new(
     ///     Point::new(-0.13806939125061035, 51.51989264641164),
     ///     Point::new(-0.13427138328552246, 51.52319711775629),
     /// );
-    /// let res = oc.forward_full(&address, &Some(bbox.into())).unwrap();
+    /// let res = oc.forward_full(&address, bbox).unwrap();
     /// let first_result = &res.results[0];
     /// // the first result is correct
     /// assert_eq!(first_result.formatted, "UCL, 188 Tottenham Court Road, London WC1E 6BT, United Kingdom");
     ///```
-    pub fn forward_full<T>(
-        &self,
-        place: &str,
-        bounds: &Option<InputBounds<T>>,
-    ) -> reqwest::Result<OpencageResponse<T>>
+    ///
+    /// ```
+    /// // You can pass NOBOX if you don't need bounds.
+    /// use geocoding::{Opencage, Point};
+    /// use geocoding::opencage::{InputBounds, NOBOX};
+    /// let oc = Opencage::new("dcdbf0d783374909b3debee728c7cc10".to_string());
+    /// let address = "UCL CASA";
+    /// let res = oc.forward_full(&address, NOBOX).unwrap();
+    /// let first_result = &res.results[0];
+    /// assert_eq!(
+    ///     first_result.formatted,
+    ///     "UCL, 188 Tottenham Court Road, London WC1E 6BT, United Kingdom"
+    /// );
+    /// ```
+    ///
+    /// ```
+    /// // There are several ways to construct a Point – the Turbofish is necessary
+    /// use geocoding::{Opencage, Point};
+    /// use geocoding::opencage::InputBounds;
+    /// let oc = Opencage::new("dcdbf0d783374909b3debee728c7cc10".to_string());
+    /// let address = "UCL CASA";
+    /// let bbox = InputBounds::new::<Point<f64>>(
+    ///     (-0.13806939125061035, 51.51989264641164).into(),
+    ///     (-0.13427138328552246, 51.52319711775629).into(),
```

Comment:
since `new` already calls [`into()`](https://github.com/georust/geocoding/pull/15/files#diff-f9842fcb9383ac44d829deb2b5e29bf2R611), do we need to call `into()` here?

Reply:
Good catch – `typeck` wanted it because I'd specified the explicit types with the turbofish.